### DISCUSSION TITLE
Design to address Restore on deactivate restore wrong state #88

### DISCUSSION THREAD

**Discussion by cayossarian** [hugobloem/stateful_scenes#155] [Ideas]
@hugobloem, I have a partial working branch where I populate a select drop down for internal 'off' scenes as part of hub.  It appears that the hub is always required and one cannot add external stand-alone scenes without first configuring a hub (as the scene.yaml path is enforced)

My assumption is then that the user would potentially pick another external scene as the 'off' scene and not some internal scene.  Thus the 'off' select would only consider other external scenes except the current one.  Are my assumptions, correct?  

So to summarize the questions I have:

1. Are there any other uses for the hubs, e.g., would one ever have two hubs (don't think the config supports this case right now).  
2. Is it desirable to link a hub/internal scene to anything like an external scene or is it sufficient to link internal scenes to internal 'off' scenes?
3. Likewise, is it necessary to define an internal scene as an 'off' scene for an external scene?

**Comment by cayossarian** [comment]
The [pull request](https://github.com/hugobloem/stateful_scenes/pull/158) assumes 'off' scene configurations will be configured for hubs/HA scenes and external scenes sepaerately.  In other words if you are configuring an external scene's 'off' state another external scene could show up in the select to 'turn off' or revert.   For external scenes it is possible to allow an internal scene but I think the intention is to undo the current scene 'on' state.  For me providing a narrow select list seemed to be more usable.  If there is another use case, please speak up.
